app/src/hooks/useTheme.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+
+import { getDarkMode } from '@utils/storage/pomodoro'
+
+import { DEFAULT_THEME, THEME_TOGGLE_TEXT, THEMES_TEXT } from '@molecules/ThemeToggle/constants'
+import type { ThemeType } from '@molecules/ThemeToggle/types'
+
+const useTheme = () => {
+  const darkModeValue = getDarkMode()
+
+  const [darkMode, setDarkMode] = useState<ThemeType>(darkModeValue) ?? DEFAULT_THEME
+  const [hasMounted, setHasMounted] = useState(false)
+
+  const toggleTheme = () => {
+    setDarkMode((prevTheme) => !prevTheme)
+  }
+  const switchLabel = !darkMode ? THEME_TOGGLE_TEXT.light : THEME_TOGGLE_TEXT.dark
+
+  useEffect(function checkMount() {
```

Comment:
This can be extracted into a custom hook.
We can install `react-use` and use their [useMount hook](https://github.com/streamich/react-use/blob/master/docs/useMount.md).